resilience4j-retry/src/main/java/io/github/resilience4j/retry/internal/RetryImpl.java

```diff
@@ -52,20 +48,21 @@
     private LongAdder failedAfterRetryCounter;
     private LongAdder succeededWithoutRetryCounter;
     private LongAdder failedWithoutRetryCounter;
-    /*package*/ static CheckedConsumer<Long> sleepFunction = Thread::sleep;
+    private CheckedConsumer<Long> sleepFunction;
 
-    public RetryImpl(String name, RetryConfig config){
+    public RetryImpl(String name, RetryConfig config) {
         this.name = name;
         this.config = config;
         this.maxAttempts = config.getMaxAttempts();
         this.intervalFunction = config.getIntervalFunction();
         this.exceptionPredicate = config.getExceptionPredicate();
-        this.metrics = this.new RetryMetrics();
         this.eventProcessor = new RetryEventProcessor();
-        succeededAfterRetryCounter = new LongAdder();
-        failedAfterRetryCounter = new LongAdder();
-        succeededWithoutRetryCounter = new LongAdder();
-        failedWithoutRetryCounter = new LongAdder();
+        this.sleepFunction = config.getSleepFunction();
```

Comment:
I think we should actually embed all this fields to `RetryMetricsImpl` completely and provide methods to update them like `RetryMetricsImpl.failedWithoutRetry()` etc.

Reply:
I'm not sure that IOC principle is actually applies here.
From my perspective all this counters don't bring any behaviour to this Retry implementation.
Actually you can completely delete remove them and nothing should actually change.

That is why I think that this counters don't belong here and should be isolated behind some other class.